Split customizer class

jonathanbardo opened the pull request:
@fjarrett Let me know what you think so far.

Commit:
Customizer component - font

Commit:
Add color class

Commit:
Fix phpcs

frankiejarrett commented on inc/customizer/colors.php:
```diff
@@ -0,0 +1,457 @@
+	 *
+	 * @var array
+	 */
+	public $color_schemes = array();
```
@jonathanbardo Maybe the visibility on these should be `protected` and we expose a magic getter method? This would make the filter the only way the values can be altered outside of the class.

jonathanbardo replied:
Yes good call 😉

Commit:
Protect all the thing

fjarrett-godaddy committed:
Support multiple font types with ease

fjarrett-godaddy committed:
Make secondary text CSS rules match across colors/fonts

fjarrett-godaddy committed:
Link color selector tweak

fjarrett-godaddy committed:
Magic property getters

fjarrett-godaddy committed:
Rename font.php to fonts.php

frankiejarrett commented on inc/customizer.php:
```diff
@@ -3,218 +3,42 @@
+
+			}
+
+			$this->$basename = new $classname;
```
@jonathanbardo What if we just require the file and it instantiates itself, like this `Primer_Customizer` class does? Also I'm unsure what `$this->$basename` is?

frankiejarrett replied:
OK I see what `$this->$basename` is now. That was tricky.

frankiejarrett replied:
Took a stab at this in https://github.com/godaddy/wp-primer-theme/commit/ca8365803a28ff0ca2483fcb958363f3c5af55e9 and refined things a bit in in https://github.com/godaddy/wp-primer-theme/pull/23/commits/9d1c045fee8234d2d7567188ca5aa541161639c2.

fjarrett-godaddy committed:
Remove deprecated tags

fjarrett-godaddy committed:
Autload customizer components, access colors with global

fjarrett-godaddy committed:
Update fonts, alphabetize

fjarrett-godaddy committed:
Clean up global usage to prevent errors

jonathanbardo commented on inc/customizer.php:
```diff
@@ -86,7 +80,7 @@ public function header() {
 			array(
 				'default-image'      => '',
-				'default-text-color' => $this->colors->get_default_color_hex( 'header_textcolor', 'default' ),
+				'default-text-color' => $GLOBALS['primer_customizer_colors']->get_default_color( 'header_textcolor', 'default' ),
```
@fjarrett What's the reasoning here? Let's avoid global if we can.

jonathanbardo commented on inc/customizer.php:
```diff
@@ -22,42 +29,24 @@ public function __construct() {
+		if ( isset( $GLOBALS['primer_customizer_colors'] ) ) {
 
-		add_theme_support( 'custom-logo', $args );
+			$this->colors = $GLOBALS['primer_customizer_colors'];
```
@fjarrett If the following filters need the customizer colors class then they should be moved to that class I think. Removing circular dependencies and globals if we can.

jonathanbardo commented:
@fjarrett One of the reason I exposed a filter and I didn't instanciate class into their own file was to make sure that a child theme could easily override and extend the fonts class or the colors class. If we instanciate in the file directly there is no way for a child theme to do that.

Commit:
Remove global usage and move method into color class

Commit:
Remove blank line after control structure

Commit:
Removing unused var + replacing output var with buffer

tobeycodes committed:
Add missing styles from #24

tobeycodes committed:
Refactored header styles.

- Row is now on the header rather than site-title, so you can now easily have a site title and menu where as before you were restricted to having a site title taking up the entire header.
- Moved featured-content to correct file

fjarrett-godaddy committed:
Adjust media query ranges to work in Customizer

fjarrett-godaddy committed:
Support RGBA CSS in the Customizer

fjarrett-godaddy committed:
Use background-color property for button hovers

fjarrett-godaddy committed:
Functions must be called first for rem-calc to work in variables

fjarrett-godaddy committed:
Fix line heights and widget font size

fjarrett-godaddy committed:
Use h4 for widget titles

fjarrett-godaddy committed:
Fix some forms CSS

fjarrett-godaddy committed:
Fix hover CSS on menu sub nav

tobeycodes committed:
Put social navigation in seperate file for child themes.

tobeycodes committed:
Add apply_filters to register_nav_menus

fjarrett-godaddy committed:
Back compat fix for less than WP 4.6

fjarrett-godaddy committed:
Fix sticky posts styles
